planner/cascades/transformation_rules.go

```diff
@@ -1203,6 +1207,94 @@ func (r *MergeAggregationProjection) OnTransform(old *memo.ExprIter) (newExprs [
 	return []*memo.GroupExpr{newAggExpr}, true, false, nil
 }
 
+// EliminateSingleMaxMin tries to convert a single max/min to Limit+Sort operators.
+type EliminateSingleMaxMin struct {
+	baseRule
+}
+
+// NewRuleEliminateSingleMaxMin creates a new Transformation EliminateSingleMaxMin.
+// The pattern of this rule is `max/min->X`.
+func NewRuleEliminateSingleMaxMin() Transformation {
+	rule := &EliminateSingleMaxMin{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandAggregation,
+		memo.EngineTiDBOnly,
+		memo.NewPattern(memo.OperandAny, memo.EngineTiDBOnly),
+	)
+	return rule
+}
+
+// Match implements Transformation interface.
+// Use appliedRuleSet in GroupExpr to avoid re-apply rules.
+func (r *EliminateSingleMaxMin) Match(expr *memo.ExprIter) bool {
+	agg := expr.GetExpr().ExprNode.(*plannercore.LogicalAggregation)
```

Comment:
In cascades planner, LogicalAggregation contains three modes: Complete, Final and Partial. I think this rule only works on the Complete mode. Also add a check of the AggMode by adding a method `IsCompleteModeAgg()`, you can refer the method `LogicalAggregation.IsPartialModeAgg()`.